src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
The main problem is, that the CMD is a builder and therefore should be call independent. 
Let me explain the main issue i have by implementing the logic directly in the CreateContainerCmd#withLinks method. 

```js
//will correctly create links on myNet
client.createContainerCmd(...)
  .withNetworkMode("myNet")
  .withLinks(...)

//will faulty create links on the legacy bridge
client.createContainerCmd(...)
  .withLinks(...)
  .withNetworkMode("myNet")
```

The issue is that the decision can't be made in the #withLinks method, because we do not know which NetworkMode is finally used.